src/kaltura-player.js

```diff
@@ -45,7 +46,11 @@ class KalturaPlayer extends FakeEventTarget {
     this._localPlayer = loadPlayer(noSourcesOptions);
     this._logger = getLogger('KalturaPlayer' + Utils.Generator.uniqueId(5));
     this._uiWrapper = new UIWrapper(this, options);
-    this._provider = new Provider(options.provider, __VERSION__);
+    if (providers.exists(options.provider.type)) {
```

Comment:
Can we make `options.provider.type` optional and take the type from the registred provider itself (this._provider.type)?

Reply:
good idea, but I think it will make this harder for use cases where someone copies SaaS player and tries to use it locally.